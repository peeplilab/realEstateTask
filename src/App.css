.inputs {
    display: flex;
    align-items: center;
}

.date-input {
    margin-right: 20px;
    height: 30px;
}

.select-input {
    width: 200px;
}
.logid{
    display: 'flex';
    justify-content: 'flex-start';
    align-items: 'center';
    flex-direction: 'row';
    width: '20vw';
    padding: 5;
}
.button{
    background-color: #4CAF50; /* Green */
    border: none;
    color: white;
    padding: 15px 18px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
  
}