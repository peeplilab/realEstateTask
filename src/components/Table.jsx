import { useEffect, useMemo, useState } from "react";
import { COLUMNS } from "../column";
import { useTable, useSortBy, useGlobalFilter, useFilters, usePagination } from "react-table";
import { ColumnFilter } from "./ColumnFilter";
import DatePicker from 'react-datepicker';
import Select from 'react-select';
import { format, isAfter, parse, parseISO, isBefore,isEqual } from 'date-fns';
import '../App.css';

import 'react-datepicker/dist/react-datepicker.css';
import { useQueryState } from "./useQueryParams";
//import { isEqual } from "date-fns/esm";

const options = [
  { value: 'LEASE_CLOSURE', label: 'LEASE_CLOSURE' },
  { value: 'ADD_COMPANY_EMPLOYEE', label: 'ADD_COMPANY_EMPLOYEE' },
  { value: 'CERT_TITLE_DEED_PLOT', label: 'CERT_TITLE_DEED_PLOT' },
  { value: 'ADD_POA', label: 'ADD_POA' },
  { value: 'ADD_COMPANY', label: 'ADD_COMPANY' },
  { value: 'CERT_PROP_OWNERSHIP', label: 'CERT_PROP_OWNERSHIP' },
  { value: 'LEASE_REGISTRATION', label: 'LEASE_REGISTRATION' },

];
const options1 = [
  { value: 'INITIATE_APPLICATION', label: 'INITIATE_APPLICATION' },
  { value: 'SUBMIT_APPLICATION', label: 'SUBMIT_APPLICATION' },
  { value: 'ADD_EMPLOYEE', label: 'ADD_EMPLOYEE' },
  { value: 'DARI_REFRESH_TOKEN', label: 'DARI_REFRESH_TOKEN' },
  { value: 'DARI_APP_LOGIN', label: 'DARI_APP_LOGIN' },

];
const Table = ({ mockData }) => {
  const columns = useMemo(() => COLUMNS, []); // memoize before adding to useTable hook
  const [fromDate, setFromDate] = useQueryState('fromDate');
  const [toDate, setToDate] = useQueryState('toDate');
  const [option, setOption] = useQueryState('actionType');
  const [logId, setLogId] = useQueryState('logId');
  const [applicationType, setApplicationType] = useQueryState('applicationType');
  const [appId, setAppId] = useQueryState('appId');
  const [filteredData, setfilteredData] = useState(mockData);
  useEffect(() => {
    setfilteredData(mockData)
  }, [mockData])

  const handleSelect = (option) => {
    setOption(option?.value || null);
  };
  const handleSelect1 = (option) => {
    setApplicationType(option?.value || null);
  };

  const handleDate = (date) => {
    setFromDate(format(date, 'MM-dd-yyyy'));
  };
  const handleDate1 = (date) => {
    setToDate(format(date, 'MM-dd-yyyy'));
  };
  // default column component
  const defaultColumn = useMemo(() => {
    return {
      Filter: ColumnFilter,
    };
  }, []);
  const handleSubmit = (event) => {
    event.preventDefault();
    let filters = {}
    let results = filteredData;
    if (logId) {
      filters.logId = logId
      results = results.filter((key) => {
        return Number(key.logId) === Number(logId)
      });
    }
    if (appId) {
      filters.appId = appId;
      results = results.filter((key) => {
        return Number(key.applicationId) === Number(appId)
      });

    }
    if (option) {
      filters.actionType = option;
      results = results.filter((key) => {
        return String(key.actionType) === String(option)
      });

    }
    if (applicationType) {
      filters.appType = applicationType;
      results = results.filter((key) => {
        return String(key.applicationType) === String(applicationType)
      });
    }
    if (fromDate) {
      let formattedFromDate = new Date(fromDate);
      results = results.filter((key) => {
        return isAfter(parseISO(key.creationTimestamp), formattedFromDate) ||
          isEqual(parseISO(key.creationTimestamp), formattedFromDate)
      });
    }
    if (toDate) {
      let formattedToDate = new Date(toDate);
      results = results.filter((key) => {
        return isBefore(parseISO(key.creationTimestamp), formattedToDate) ||
          isEqual(parseISO(key.creationTimestamp), formattedToDate)
      });
    }
    setfilteredData(results);
    return results;

  }

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    prepareRow,    
    page,
    canPreviousPage,
    canNextPage,
    pageOptions,
    pageCount,
    gotoPage,
    nextPage,
    previousPage,
    setPageSize,
    state: { pageIndex, pageSize },
  } = useTable(
    {
      columns,
      data: filteredData,
      defaultColumn,
    },

    useFilters,
    useGlobalFilter,
    useSortBy,
    usePagination,
  );

  return (
    <>
    <div style={{alignItems:'revert'}}>
    <p>{`Home > Administration > Logger Search`}</p>
    </div>
      <div className="inputs">
        <div style={{
          marginRight: 30,
        }} className="logid">
          <input
            style={{
              height: 30,
            }}
            type="number"
            value={logId}
            onChange={(e) => setLogId(e.target.value)}
            placeholder="Log ID"
          />
        </div>
        <div className="logid" style={{
          marginRight: 30,
        }} >
          <input
            style={{ height: 30 }}
            type="number"
            value={appId}
            onChange={(e) => setAppId(e.target.value)}
            placeholder="Application Id"
          />
        </div>
        <div>
          <Select
            className="select-input"
            isClearable
            value={options1.find((o) => o.value === applicationType)}
            onChange={handleSelect1}
            options={options}
            placeholder="Application Type"
          />
        </div>
      </div>
      <div className="inputs">
        <div style={{
          marginRight: 30,
        }} >
          <Select
            className="select-input"
            isClearable
            value={options.find((o) => o.value === option)}
            onChange={handleSelect}
            options={options1}
            placeholder="Action Type"
          />
        </div>
        <div>
          <DatePicker
            className="date-input"
            selected={fromDate ? parse(fromDate, 'MM-dd-yyyy', new Date()) : new Date()}
            onChange={handleDate}
          />
        </div>
        <div>
          <DatePicker
            className="date-input"
            selected={toDate ? parse(toDate, 'MM-dd-yyyy', new Date()) : new Date()}
            onChange={handleDate1}
          />
        </div>
        <button
          onClick={(e) => handleSubmit(e)}
          className='button'
        >
          Search
        </button>

      </div>
      {/* <GlobalFilter globalFilter={globalFilter} setGlobalFilter={setGlobalFilter} /> */}

      {/* apply the table props */}
      <table {...getTableProps()}>
        <thead>
          {
            // Loop over the header rows
            headerGroups.map((headerGroup) => (
              // Apply the header row props
              <tr {...headerGroup.getHeaderGroupProps()}>
                {
                  // Loop over the headers in each row
                  headerGroup.headers.map((column) => (
                    // Apply the header cell props
                    <th {...column.getHeaderProps(column.getSortByToggleProps())}>
                      {column.render("Header")}
                      <span>{column.isSorted ? (column.isSortedDesc ? " ðŸ”½" : " ðŸ”¼") : ""}</span>
                      <div>{column.canFilter ? column.render("Filter") : null}</div>
                    </th>
                  ))
                }
              </tr>
            ))
          }
        </thead>
        {/* Apply the table body props */}
        <tbody {...getTableBodyProps()}>
          {
            // Loop over the table rows
            page.map((row, i) => {
              // Prepare the row for display
              prepareRow(row);
              return (
                <tr {...row.getRowProps()}>

                  {
                    // Loop over the rows cells
                    row.cells.map((cell) => {
                      // Apply the cell props
                      return <td {...cell.getCellProps()}>{cell.render("Cell")}</td>;
                    })
                  }
                </tr>
              );
            })
          }
        </tbody>
      </table>

      <div className="pagination" style={{ marginTop: "1rem" }}>
        <button onClick={() => gotoPage(0)} disabled={!canPreviousPage}>
          {"<<"}
        </button>{" "}
        <button onClick={() => previousPage()} disabled={!canPreviousPage}>
          {"<"}
        </button>{" "}
        <button onClick={() => nextPage()} disabled={!canNextPage}>
          {">"}
        </button>{" "}
        <button onClick={() => gotoPage(pageCount - 1)} disabled={!canNextPage}>
          {">>"}
        </button>{" "}
        <span>
          Page{" "}
          <strong>
            {pageIndex + 1} of {pageOptions.length}
          </strong>{" "}
        </span>
        <span>
          | Go to page:{" "}
          <input
            type="number"
            defaultValue={pageIndex + 1}
            onChange={(e) => {
              const page = e.target.value ? Number(e.target.value) - 1 : 0;
              gotoPage(page);
            }}
            style={{ width: "100px" }}
          />
        </span>{" "}
        <select
          value={pageSize}
          onChange={(e) => {
            setPageSize(Number(e.target.value));
          }}
        >
          {[10, 20, 30, 40, 50].map((pageSize) => (
            <option key={pageSize} value={pageSize}>
              Show {pageSize}
            </option>
          ))}
        </select>
      </div>
    </>
  );
};

export default Table;
