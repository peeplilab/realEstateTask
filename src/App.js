import React, { useEffect, useState } from "react";
import Table from "./components/Table";
import axios from "axios";

function App() {
  const [data, setData] = useState([]);

  const fetchData = async () => {
    const { data } = await axios("https://run.mocky.io/v3/a2fbc23e-069e-4ba5-954c-cd910986f40f").catch((err) => console.log(err));
    setData(data.result.auditLog);
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div className="container">
      <Table mockData={data} />
    </div>
  );
}

export default App;
