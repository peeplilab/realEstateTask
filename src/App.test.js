import { render, screen } from '@testing-library/react'
import App from './App'
import Table from './components/Table'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders learn text', () => {
  const tree = renderer
    .create(
      <Router>
        <App />
      </Router>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
