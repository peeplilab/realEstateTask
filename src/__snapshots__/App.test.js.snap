// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn text 1`] = `
<div
  className="container"
>
  <div
    style={
      Object {
        "alignItems": "revert",
      }
    }
  >
    <p>
      Home &gt; Administration &gt; Logger Search
    </p>
  </div>
  <div
    className="inputs"
  >
    <div
      className="logid"
      style={
        Object {
          "marginRight": 30,
        }
      }
    >
      <input
        onChange={[Function]}
        placeholder="Log ID"
        style={
          Object {
            "height": 30,
          }
        }
        type="number"
      />
    </div>
    <div
      className="logid"
      style={
        Object {
          "marginRight": 30,
        }
      }
    >
      <input
        onChange={[Function]}
        placeholder="Application Id"
        style={
          Object {
            "height": 30,
          }
        }
        type="number"
      />
    </div>
    <div>
      <div
        className="select-input css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-13cymwt-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-1fdsijx-ValueContainer"
          >
            <div
              className=" css-1jqq78o-placeholder"
              id="react-select-2-placeholder"
            >
              Application Type
            </div>
            <div
              className=" css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-1xc3v61-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="inputs"
  >
    <div
      style={
        Object {
          "marginRight": 30,
        }
      }
    >
      <div
        className="select-input css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-13cymwt-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-1fdsijx-ValueContainer"
          >
            <div
              className=" css-1jqq78o-placeholder"
              id="react-select-3-placeholder"
            >
              Action Type
            </div>
            <div
              className=" css-qbdosj-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-1xc3v61-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="date-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="01/04/2023"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="date-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="01/04/2023"
          />
        </div>
      </div>
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <table
    role="table"
  >
    <thead>
      <tr
        role="row"
      >
        <th
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          LogId
          <span />
          <div />
        </th>
        <th
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Action Type
          <span />
          <div />
        </th>
        <th
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Application Id
          <span />
          <div />
        </th>
        <th
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Application Type
          <span />
          <div />
        </th>
        <th
          colSpan={1}
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Toggle SortBy"
        >
          Date
          <span />
          <div />
        </th>
      </tr>
    </thead>
    <tbody
      role="rowgroup"
    />
  </table>
  <div
    className="pagination"
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <button
      disabled={true}
      onClick={[Function]}
    >
      &lt;&lt;
    </button>
     
    <button
      disabled={true}
      onClick={[Function]}
    >
      &lt;
    </button>
     
    <button
      disabled={true}
      onClick={[Function]}
    >
      &gt;
    </button>
     
    <button
      disabled={true}
      onClick={[Function]}
    >
      &gt;&gt;
    </button>
     
    <span>
      Page
       
      <strong>
        1
         of 
        0
      </strong>
       
    </span>
    <span>
      | Go to page:
       
      <input
        defaultValue={1}
        onChange={[Function]}
        style={
          Object {
            "width": "100px",
          }
        }
        type="number"
      />
    </span>
     
    <select
      onChange={[Function]}
      value={10}
    >
      <option
        value={10}
      >
        Show 
        10
      </option>
      <option
        value={20}
      >
        Show 
        20
      </option>
      <option
        value={30}
      >
        Show 
        30
      </option>
      <option
        value={40}
      >
        Show 
        40
      </option>
      <option
        value={50}
      >
        Show 
        50
      </option>
    </select>
  </div>
</div>
`;
